lib/hedwig_sms.ex

```diff
@@ -8,7 +8,7 @@ defmodule Hedwig.Adapters.SMS do
   @doc false
   def init({robot, opts}) do
     HTTPoison.start
-    Hedwig.Robot.register(robot, opts[:name])
+    :global.register_name(opts[:name], robot)
```

Comment:
What if instead of registering the pid of the robot, I register the pid of the adapter. That way I could have a function on the adapter that calls it to get the state containing the robot pid and do what is needed from there?